edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.Test/DockerConfigTest.cs

```diff
@@ -124,5 +138,51 @@ public void TestSerialization()
             Assert.Equal(expected, json);
         }
+
+        [Theory]
+        [InlineData(null, null, typeof(ArgumentException))]
+        [InlineData("", null, typeof(ArgumentException))]
+        [InlineData("  ", null, typeof(ArgumentException))]
+        [InlineData("mcr.ms.com/ea:t1:t2", null, typeof(ArgumentException))]
+        [InlineData("ea:t1:t2", null, typeof(ArgumentException))]
+        [InlineData("ea:t1", "ea:t1", null)]
+        [InlineData(" ea:t1 ", "ea:t1", null)]
+        [InlineData("mcr.ms.com/ea:t1", "mcr.ms.com/ea:t1", null)]
+        [InlineData("mcr.ms.com/ea", "mcr.ms.com/ea:latest", null)]
+        [InlineData(" ubuntu ", "ubuntu:latest", null)]
+        public void TestValidateAndGetImage(string image, string result, Type expectedException)
+        {
+            if (expectedException != null)
+            {
+                Assert.Throws(expectedException, () => DockerConfig.ValidateAndGetImage(image));
+            }
+            else
+            {
+                string updatedImage = DockerConfig.ValidateAndGetImage(image);
+                Assert.Equal(result, updatedImage);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTestGetCreateOptionsData))]
+        public void TestGetCreateOptions(string createOptions, CreateContainerParameters expectedCreateOptions)
+        {
+            CreateContainerParameters createContainerParameters = DockerConfig.GetCreateOptions(createOptions);
+
+            Assert.NotNull(createContainerParameters);
+            Assert.True(DockerConfig.CompareCreateOptions(expectedCreateOptions, createContainerParameters));
+        }
+
+        static IEnumerable<object[]> GetTestGetCreateOptionsData()
```

Comment:
please make this as public; I have PR to upgrade XUnit NuGet package to v2.4.1, which enforces all data provider method should be public.